src/main/java/seedu/address/storage/JsonAdaptedMedication.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.storage;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.record.Medication;
+
+/**
+ * Jackson-friendly version of {@link Medication}.
+ */
+class JsonAdaptedMedication {
+
+    private final String medicationName;
+
+    /**
+     * Constructs a {@code JsonAdaptedMedication} with the given {@code medicationName}.
+     */
+    @JsonCreator
+    public JsonAdaptedMedication(String medicationName) {
+        this.medicationName = medicationName;
+    }
+
+    /**
+     * Converts a given {@code Medication} into this class for Jackson use.
+     */
+    public JsonAdaptedMedication(Medication source) {
+        medicationName = source.medicationName;
+    }
+
+    @JsonValue
+    public String getMedicationName() {
+        return medicationName;
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted tag object into the model's {@code Medication} object.
```

Comment:
May want to update "tag object" to medication object for continuity. Similarly for line 39.

Reply:
Alright! Will amend